sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -82,6 +106,14 @@ object AlluxioUtils {
         // replace all of rootPaths which are already unique
         val rootPaths = relation.location.rootPaths.map(replaceFunc)
 
+        // check the alluxio paths in root paths exist or not
+        // throw out an exception to stop the job when any of them is not mounted
```

Comment:
we are having to iterate thought the paths twice, can't we just check when we go to replace it in replaceFunc?  

In some ways it would also be nice to see track of if we replaced any and if not we just skip the next inferring logic, but that could be separate.

Reply:
replaceFunc will also be used in inferPartitioning. There will go through all the inputFiles and call replaceFunc to check.
If we want to go one iteration, I'll need to have another function variable to complete replaceFunc and checkAlluxioMounted together. Not sure if it's worth to be optimized because the root paths should be short enough.

Reply:
ok this is fine for now